### PR TITLE
Fix/update e2e tests to support OpenShift environment

### PR DESCRIPTION
**Pull request by jkandasa**
# Changes
* modified e2e tests and added additional tests in TektonConfigDeployment, TektonHubDeployment
* modified TektonChainTutorial e2e tests, to support in OpenShift environment
* modified TektonHub reconciler to address some of issues
* added ClusterRole, ClusterRoleBinding, ServiceAccount informers into InstallerSet reconciler, without this service account is not created occasionally. 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by jkandasa** [comment]
cc @concaf @PuneetPunamiya @vdemeester

File: go.mod [hunk 1]
```diff
@@ -19,6 +19,7 @@ require (
 	github.com/openshift/client-go v0.0.0-20210521082421-73d9475a9142
 	github.com/sigstore/cosign/v2 v2.0.0
 	github.com/spf13/viper v1.15.0
+	github.com/stretchr/testify v1.8.1
```
Comment (vdemeester) [inline go.mod:22]: We are using `gotest.tools` for assertions and stuff (`gotest.tools/v3 v3.4.0`), so if we could stick to it that would be nice(r) 👼🏼

Comment (jkandasa) [reply go.mod:22]: @vdemeester I think I can switch to `gotest.tools` for assertions easily.

Shall I continue with `github.com/stretchr/testify` for `suite interface`? 
do we have a similar concept in `gotest.tools`?

Comment (vdemeester) [reply go.mod]: @jkandasa we don't, but as discussed offline, it's fine to use testify, let's not block on this.

**Comment by piyush-garg** [comment]
i am merging this PR for now

we will keep adding and refactoring in tests as we encounter anything.

/lgtm

**Pull request merged**